src/utils/schemaUtils.js

```diff
@@ -25,8 +25,53 @@ function getAllResponseContentTypes(responses) {
     return Array.from(resultSet);
 }
 
+/**
+ * recursively omit prop with the given name and if it equals to the given value
+ *
+ * @param {object} dereferencedSchema dereferenced schema
+ * @param {string} omitByPropName the prop name to omit
+ * @param {string} omitByValue omit if the prop value equals to this value
+ */
+function omitPropsFromSchema(dereferencedSchema, omitByPropName, omitByValue) {
+    if (dereferencedSchema.type === 'object') {
+        const newSchema = { ...dereferencedSchema };
+        const schemaProperties = dereferencedSchema.properties;
+        const newSchemaProperties = { ...schemaProperties };
+        for (const propName of Object.keys(newSchemaProperties)) {
+            if (newSchemaProperties[propName][omitByPropName] === omitByValue) {
+                // delete the prop from properties object so it would be accepted in case of additionalProperties: true
+                delete newSchemaProperties[propName];
+
+                // delete the prop from the required props
+                const index = newSchema.required ? newSchema.required.indexOf(propName) : -1;
+                if (index >= 0) {
+                    newSchema.required = newSchema.required.slice(0, index)
+                        .concat(newSchema.required.slice(index + 1));
+                }
+            } else if (newSchemaProperties[propName].type === 'object') {
+                // if the current prop is an object we need to recursively look for omitByPropName occurrences
+                newSchemaProperties[propName] = omitPropsFromSchema(newSchemaProperties[propName], omitByPropName, omitByValue);
+            }
+        }
+        newSchema.properties = newSchemaProperties;
+        return newSchema;
+    } else if (dereferencedSchema.type === 'array' && dereferencedSchema.items.type === 'object') {
+        const newSchema = { ...dereferencedSchema };
+        newSchema.items = omitPropsFromSchema(dereferencedSchema.items, omitByPropName, omitByValue);
+        return newSchema;
+    } else {
+        return dereferencedSchema;
+    }
+}
+
+function isOpenApi3(dereferenced) {
+    return dereferenced.openapi ? dereferenced.openapi.startsWith('3.0') : false;
```

Comment:
Shouldn't it be just `3.`? They might publish 3.1.x at some point.

Reply:
i wasn't sure about that. i guess we should be able to support 3.x.x as it should be a breaking change.

what do you say?

Reply:
Yup, OpenAPI is following SemVer, so 3.x.x definitely shouldn't be a breaking change and we should be able to support it out-of-the-box.